### PR TITLE
[Refact](type system)refact interconversion for jsonb with column 

### PR DESCRIPTION
**Pull request by amorynan**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by amorynan** [comment]
run buildall

File: be/src/vec/jsonb/serialize.cpp [hunk 1]
```diff
@@ -17,268 +17,265 @@
...
-            LOG(FATAL)
-                    << "FieldType::OLAP_FIELD_TYPE_DECIMAL not implemented use DecimalV3 instead";
-            break;
-        default:
-            LOG(FATAL) << "unknow type " << int(tablet_column.type());
-            break;
-        }
-    }
-}
+//static inline bool is_column_null_at(int row, const IColumn* column, const doris::FieldType& type,
```
Comment (eldenmoon) [inline be/src/vec/jsonb/serialize.cpp:27]: remove unused code

File: be/src/vec/data_types/serde/data_type_array_serde.h [hunk 2]
```diff
@@ -32,6 +33,11 @@ class DataTypeArraySerDe : public DataTypeSerDe {
         LOG(FATAL) << "Not support read from pb to array";
     }
 
+    Status write_column_to_jsonb(const IColumn& column, JsonbWriter& result, Arena* mem_pool,
```
[the remark this answers is not in the document]
Comment (eldenmoon) [reply be/src/vec/data_types/serde/data_type_array_serde.h:36]: I think we should not return any status for this function

**Review by eldenmoon — CHANGES_REQUESTED** [review]

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.3 seconds
 stream load tsv:          428 seconds loaded 74807831229 Bytes, about 166 MB/s
 stream load json:         24 seconds loaded 2358488459 Bytes, about 93 MB/s
 stream load orc:          59 seconds loaded 1101869774 Bytes, about 17 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230419133438_clickbench_pr_131103.html

File: be/src/vec/data_types/serde/data_type_array_serde.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (amorynan) [reply be/src/vec/data_types/serde/data_type_array_serde.h:36]: U mean Interface ?

File: be/src/vec/data_types/serde/data_type_array_serde.h [hunk 3]
```diff
@@ -32,6 +33,11 @@ class DataTypeArraySerDe : public DataTypeSerDe {
...
+                                 const int32_t col_id, const int row_num) const override;
```
[the remark this answers is not in the document]
Comment (amorynan) [reply be/src/vec/data_types/serde/data_type_array_serde.h:37]: is const Just make others who may write other data type serde function known this col _id can not changed ?

File: be/src/vec/data_types/serde/data_type_number_serde.h [hunk 4]
```diff
@@ -157,5 +162,62 @@ Status DataTypeNumberSerDe<T>::write_column_to_pb(const IColumn& column, PValues
     return Status::OK();
 }
 
+template <typename T>
+Status DataTypeNumberSerDe<T>::read_column_from_jsonb(IColumn& column,
+                                                      const JsonbValue* arg) const {
+    if constexpr (std::is_same_v<T, Int8>) {
+        column.insert(static_cast<const JsonbInt8Val*>(arg)->val());
```
[the remark this answers is not in the document]
Comment (amorynan) [reply be/src/vec/data_types/serde/data_type_number_serde.h:169]: emm origin code insert(Field), maybe I can use insert_value

File: be/src/vec/jsonb/serialize.cpp (hunk 1 above)
Comment (amorynan) [reply be/src/vec/jsonb/serialize.cpp:27]: done

File: be/src/vec/data_types/serde/data_type_array_serde.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (amorynan) [reply be/src/vec/data_types/serde/data_type_array_serde.h:36]: OK I will use void

**Comment by amorynan** [comment]
run buildall

**Review by yiguolei — DISMISSED** [review]
LGTM

**Review by eldenmoon — DISMISSED** [review]
LGTM

**Comment by amorynan** [comment]
run buildall

**Review by eldenmoon — DISMISSED** [review]
LGTM

**Comment by amorynan** [comment]
run arm

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run clickbench

**Comment by amorynan** [comment]
run arm

File: be/src/vec/data_types/serde/data_type_hll_serde.cpp [hunk 5]
```diff
@@ -17,16 +17,8 @@
...
-#include <gen_cpp/types.pb.h>
-#include <stddef.h>
-#include <stdint.h>
-
-#include <string>
-
-#include "olap/hll.h"
-#include "util/slice.h"
 #include "vec/columns/column_complex.h"
-#include "vec/common/assert_cast.h"
```
Comment (adonis0147) [inline be/src/vec/data_types/serde/data_type_hll_serde.cpp:29]: Don't change these headers. We should include headers directly. Transitive headers may slow down the compilation.

**Review by adonis0147 — CHANGES_REQUESTED** [review]

File: be/src/vec/jsonb/serialize.cpp [hunk 6]
```diff
@@ -17,291 +17,13 @@
...
-#include "util/slice.h"
-#include "vec/columns/column.h"
-#include "vec/columns/column_string.h"
 #include "vec/common/arena.h"
-#include "vec/common/string_ref.h"
-#include "vec/core/column_with_type_and_name.h"
-#include "vec/core/field.h"
-#include "vec/core/types.h"
-#include "vec/data_types/data_type.h"
-#include "vec/runtime/vdatetime_value.h"
```
Comment (adonis0147) [inline be/src/vec/jsonb/serialize.cpp:48]: Same reason

**Review by adonis0147 — DISMISSED** [review]

**Comment by amorynan** [comment]
run p0

**Comment by amorynan** [comment]
run p0

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run p1

**Review by yiguolei — APPROVED** [review]
lgtm

**Pull request merged**